pkgs/development/compilers/gcc/7/default.nix

```diff
@@ -260,7 +261,12 @@ stdenv.mkDerivation ({
     export LDFLAGS_FOR_TARGET="-Wl,-rpath,$prefix/lib/amd64 $LDFLAGS_FOR_TARGET"
     export CXXFLAGS_FOR_TARGET="-Wl,-rpath,$prefix/lib/amd64 $CXXFLAGS_FOR_TARGET"
     export CFLAGS_FOR_TARGET="-Wl,-rpath,$prefix/lib/amd64 $CFLAGS_FOR_TARGET"
-  '';
+  '' + (if hostPlatform != targetPlatform && targetPlatform.isPower then ''
+    # the float128 detection in cross-gcc and glibc is a little busted. since glibc doesn't compile without float128 support, fix it.
+    sed 's/-mfloat128/-mfloat128-type -mfloat128/g' -i libgcc/configure
+    # this file also includes totally unneeded headers which are not available in cross-compilation. remove them.
+    sed -E 's/#include <(string|stdlib|ctype).h>//' -i libgcc/config/rs6000/float128-ifunc.c
```

Comment:
Probably good to put the comment outside the string, so changing the explanation doesn't cause rebuilds.

Reply:
Makes sense. So like this?

```
# the float128 detection in cross-gcc and glibc is a little busted. since glibc doesn't compile without float128 support, fix it.
+ "\nsed 's/-mfloat128/-mfloat128-type -mfloat128/g' -i libgcc/configure"
# this file also includes totally unneeded headers which are not available in cross-compilation. remove them.
+"\nsed -E 's/#include <(string|stdlib|ctype).h>//' -i libgcc/config/rs6000/float128-ifunc.c"
```

Anyway, it turns out this hack was also obsoleted by just using `--with-long-double-128 --enable-softfloat` and I just overlooked it when cleaning up …